sdk/src/main/java/io/radar/sdk/RadarJobScheduler.kt

```diff
@@ -64,7 +83,28 @@ class RadarJobScheduler : JobService() {
             this.time = time
         }
 
-        val sourceStr = extras.getString(EXTRA_SOURCE) ?: return false
+        val sourceStr = extras.getString(EXTRA_SOURCE)
+        if (Radar.isTestKey()) {
+            val batteryState = Radar.batteryManager.getBatteryState()
+            Radar.logger.d(
+                "Starting Location Job | " +
+                        "location = $location; " +
+                        "source = ${sourceStr}; " +
+                        "requestNumber = ${counter.get()}; " +
+                        "standbyBucket = ${Radar.batteryManager.getAppStandbyBucket()}; " +
+                        "performanceState = ${batteryState.performanceState.name}; " +
+                        "isCharging = ${batteryState.isCharging}; " +
+                        "batteryPercentage = ${batteryState.percent}; " +
+                        "isPowerSaveMode = ${batteryState.powerSaveMode}; " +
+                        "isIgnoringBatteryOptimizations = ${batteryState.isIgnoringBatteryOptimizations}; " +
+                        "locationPowerSaveMode = ${batteryState.getPowerLocationPowerSaveModeString()}; " +
+                        "isDozeMode = ${batteryState.isDeviceIdleMode}"
```

Comment:
I'm also wondering if we want to include these battery manager states on every log - how often do they really change? and OOMs could occur faster with longer strings

Reply:
Right now this is still all behind the test-key check. I don't think String size is much of a concern here, however I have some notes elsewhere about how we can reduce this payload size. I think the purpose of this is to understand how often these modes change, and maybe reduce it afterwards. I'd agree that these different performance modifiers probably change at different times, and that there may be a more optimal log timing.